src/shogun/classifier/AveragedPerceptron.h

```diff
@@ -35,12 +36,20 @@ class CAveragedPerceptron : public CLinearMachine
 		/** default constructor */
 		CAveragedPerceptron();
 
+		CAveragedPerceptron(CDotFeatures* traindat, CLabels* trainlab);
```

Comment:
No need to add those ctors really as it all will be done via default ctor and then put calls

Reply:
I did the required changes, and brought back the assertions. Averaged perceptron is pretty much the same as normal perceptron (except for averaging), so I imitated what was done in CPerceptron class. I think it can benefit from the changes you mentioned (using linalg and putting assertions), I can do them after this request.